src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -252,7 +252,11 @@ pub fn compile(sess: &ParseSess, features: &RefCell<Features>, def: &ast::Item)
         valid: valid,
     });
 
-    NormalTT(exp, Some(def.span), attr::contains_name(&def.attrs, "allow_internal_unstable"))
+    NormalTT(
+             exp,
+             Some((def.id, def.span)),
+             attr::contains_name(&def.attrs, "allow_internal_unstable")
+    )
```

Comment:
nit: nonstandard formatting -- should be three lines with "visual indenting" or indented one block with four/five lines.